From 4dece111fc267b7a7a6f9b87ef42026ea128758b Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Fri, 26 Apr 2024 09:55:54 +0200
Subject: [PATCH 0817/1695] treewide: st: remove deprecated pins-are-numbered
 in device tree

Align the binding and the stm32mp device tree with Linux kernel,
remove the deprecated properties pins-are-numbered.

No functional impact as it is not used in code.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 798a167930be4e82df9da6c43a7757abf4e6f19e treewide: st: remove deprecated pins-are-numbered in device tree]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I2a835d62c2207ca041fd06d62608dd0dc752dbed
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/376735
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp131.dtsi                             | 1 -
 core/arch/arm/dts/stm32mp151.dtsi                             | 2 --
 core/arch/arm/dts/stm32mp251.dtsi                             | 2 --
 .../devicetree/bindings/pinctrl/st,stm32-pinctrl.yaml         | 4 +++-
 4 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp131.dtsi b/core/arch/arm/dts/stm32mp131.dtsi
index cbefcc856..a72a75dbe 100644
--- a/core/arch/arm/dts/stm32mp131.dtsi
+++ b/core/arch/arm/dts/stm32mp131.dtsi
@@ -466,7 +466,6 @@
 			#size-cells = <1>;
 			compatible = "st,stm32mp135-pinctrl";
 			ranges = <0 0x50002000 0x8400>;
-			pins-are-numbered;
 
 			gpioa: gpio@50002000 {
 				gpio-controller;
diff --git a/core/arch/arm/dts/stm32mp151.dtsi b/core/arch/arm/dts/stm32mp151.dtsi
index b4aee1d19..0d8683bd5 100644
--- a/core/arch/arm/dts/stm32mp151.dtsi
+++ b/core/arch/arm/dts/stm32mp151.dtsi
@@ -1747,7 +1747,6 @@
 			ranges = <0 0x50002000 0xa400>;
 			interrupt-parent = <&exti>;
 			st,syscfg = <&exti 0x60 0xff>;
-			pins-are-numbered;
 
 			gpioa: gpio@50002000 {
 				gpio-controller;
@@ -1876,7 +1875,6 @@
 			#size-cells = <1>;
 			compatible = "st,stm32mp157-z-pinctrl";
 			ranges = <0 0x54004000 0x400>;
-			pins-are-numbered;
 			interrupt-parent = <&exti>;
 			st,syscfg = <&exti 0x60 0xff>;
 
diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 6dca04d18..069beb91c 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -911,7 +911,6 @@
 			#size-cells = <1>;
 			compatible = "st,stm32mp257-pinctrl";
 			ranges = <0 0x44240000 0xa0400>;
-			pins-are-numbered;
 
 			gpioa: gpio@44240000 {
 				gpio-controller;
@@ -1085,7 +1084,6 @@
 			#size-cells = <1>;
 			compatible = "st,stm32mp257-z-pinctrl";
 			ranges = <0 0x46200000 0x400>;
-			pins-are-numbered;
 
 			gpioz: gpio@46200000 {
 				gpio-controller;
diff --git a/documentation/devicetree/bindings/pinctrl/st,stm32-pinctrl.yaml b/documentation/devicetree/bindings/pinctrl/st,stm32-pinctrl.yaml
index 055cf94f2..8ffb95f63 100644
--- a/documentation/devicetree/bindings/pinctrl/st,stm32-pinctrl.yaml
+++ b/documentation/devicetree/bindings/pinctrl/st,stm32-pinctrl.yaml
@@ -36,7 +36,9 @@ properties:
     const: 1
 
   ranges: true
-  pins-are-numbered: true
+  pins-are-numbered:
+    $ref: /schemas/types.yaml#/definitions/flag
+    deprecated: true
   hwlocks: true
 
   interrupts:
-- 
2.47.2

